recipe/meta.yaml

```diff
@@ -45,36 +47,46 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != "12.0"]
+    - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
+    - cuda-version {{ cuda_major }}.0
     - patchelf >=0.12  # [linux]
     # to suppress ".so not found" errors
-    - libzlib  # [linux]
+    - libzlib       # [linux]
     - libzlib-wapi  # [win64]
   run:
+    - {{ pin_compatible("cuda-version", max_pin="x") }}
+    {% if cuda_major == "11" %}
     - cudatoolkit 11.*
+    {% else %}
+    - cuda-cudart-dev
```

Comment:
Why is the dev package required at runtime?

Reply:
Guess the issue is packages that depend on `cudnn` at runtime would now pull in these headers

Maybe we could put this in `run_constrained`? Admittedly that would serve a primarily documentation role

Alternatively we could rely on `cuda-nvcc_{{ target_platform }}` to pull in `cuda-cudart-dev`

Reply:
Sounds like dropping this is probably the consensus answer for now.

```suggestion
```

Reply:
Meant to add:

> Alternatively we could rely on cuda-nvcc_{{ target_platform }} to pull in cuda-cudart-dev

I agree this is sufficient, so I removed the dependency.